taca/illumina/NextSeq_Runs.py

```diff
@@ -1,18 +1,27 @@
 import os
+import re
+import csv
+import glob
+import shutil
+import gzip
+import operator
+import subprocess
 from datetime import datetime
-
-from flowcell_parser.classes import SampleSheetParser
-from taca.utils.filesystem import chdir
+from taca.utils.filesystem import chdir, control_fastq_filename
 from taca.illumina.Runs import Run
+from taca.illumina.HiSeqX_Runs import HiSeqX_Run
 from taca.utils import misc
+from flowcell_parser.classes import RunParametersParser, SampleSheetParser, RunParser, LaneBarcodeParser, DemuxSummaryParser
+
 
 import logging
+
 logger = logging.getLogger(__name__)
 
-class NextSeq_Run(Run):
+class NextSeq_Run(HiSeqX_Run):
```

Comment:
Just a thought. In my mind, it's not quite clear why the NextSeq class inherits from the HiSeqX class (same with NovaSeq and MiSeq). Maybe we should look at moving common classes into the Run class? That way, if we stop using HiSeqX it would be easier if we want to remove that class from the code. Just a thought though, great work Chuan! 😃

Reply:
Thanks Sara! Yes, I also thought about that. MiSeq and HiSeq 2500 have different logics of dealing with samplesheets (formats, how samplesheet interact with instrument, etc). It was from HiSeq X that things started to get "formulated". I agree that now the method of HiSeq X should be regarded as the common one and MiSeq should be a special case.

Reply:
Great! Makes sense 👍